x/checkers/keeper/msg_server_create_game.go

```diff
@@ -2,16 +2,34 @@ package keeper
 
 import (
 	"context"
+	"strconv"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	rules "github.com/xavierlepretre/checkers/x/checkers/rules"
 	"github.com/xavierlepretre/checkers/x/checkers/types"
 )
 
 func (k msgServer) CreateGame(goCtx context.Context, msg *types.MsgCreateGame) (*types.MsgCreateGameResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO: Handling the message
-	_ = ctx
+	nextGame, found := k.Keeper.GetNextGame(ctx)
+	if !found {
+		panic("NextGame not found")
+	}
+	newIndex := strconv.FormatUint(nextGame.IdValue, 10)
+	newGame := types.FullGame{
+		Creator: sdk.AccAddress(msg.Creator),
```

Comment:
this could be an invalid address, we should check validity.

Reply:
By invalid, do you mean that:

1. The bytes are malformed?
2. It has no token balance?
3. Something else?

Reply:
I as a user could submit anything as a string, either causing panics to appear later on or this tx to fail, but to the user they would not know why. 

Usually I have used `func AccAddressFromBech32(address string) (addr AccAddress, err error) {` as the way to check validity